fastapi/dependencies/utils.py

```diff
@@ -704,21 +714,41 @@ def request_params_to_args(
     return values, errors
 
 
+def _should_embed_body_fields(fields: List[ModelField]) -> bool:
+    if not fields:
+        return False
+    first_field = fields[0]
+    body_param_names_set = {field.name for field in fields}
+    embed = (
+        # A top level field has to be a single field, not multiple
+        len(body_param_names_set) > 1
+        # If it explicitly specifies it is embedded, it's not top level
+        or getattr(first_field.field_info, "embed", None)
+        # If it's a Form (or File) field, it has to be a BaseModel to be top level
+        # otherwise it has to be embedded, so that the key value pair can be extracted
+        or (
+            isinstance(first_field.field_info, params.Form)
+            and not isinstance(first_field.type_, BaseModel)
+        )
+    )
+    return embed
```

Comment:
NIT 🙈

no logic changes, just a small simplification

```suggestion
def _should_embed_body_fields(fields: List[ModelField]) -> bool:
    if not fields:
        return False
    
    # A top level field has to be a single field, not multiple
    if len(fields) > 1:
        return True
    
    first_field = fields[0]
    # If it explicitly specifies it is embedded, it's not top level
    if getattr(first_field.field_info, "embed", None):
        return True
     
    # If it's a Form (or File) field, it has to be a BaseModel to be top level
    # otherwise it has to be embedded, so that the key value pair can be extracted
    if isinstance(first_field.field_info, params.Form) and not isinstance(first_field.type_, BaseModel):
        return True
        
     return False
     
```

Reply:
Oops, my bad! 😅 The PR was still a draft. Just ignore it for 🙈

Reply:
Thanks! Good idea. I needed to tweak it a bit, but yep, it simplifies it.